lib/accessly/policy/base.rb

```diff
@@ -0,0 +1,270 @@
+module Accessly
+  module Policy
+    class Base
+
+      attr_reader :actor
+
+      def initialize(actor)
+        @actor = actor
+      end
+
+      def self.actions(actions)
+        _actions.merge!(actions)
+        actions.each do |action, action_id|
+          _define_action_methods(action, action_id)
+        end
+      end
+
+      def self.actions_on_objects(actions_on_objects)
+        _actions_on_objects.merge!(actions_on_objects)
+        actions_on_objects.each do |action, action_id|
+          _define_action_methods(action, action_id)
+        end
+      end
+
+      def self.namespace
+        String(self)
+      end
+
+      def namespace
+        self.class.namespace
+      end
+
+      def self.model_scope
+        puts caller
+        raise ArgumentError.new("#model_scope is not defined on #{self.name}.")
+      end
+
+      def model_scope
+        self.class.model_scope
+      end
+
+      def is_admin?
```

Comment:
You have a good point that we're trying not to tie the Lessonly definition of what gives an actor the rights to do whatever they want to accessly.

I'm a little hesitant to use the term `overrides` because that word to me is the programming definition, which means you're creating your own definition for something that already exists. That's not *quite* what this method is for, though - it's to give the actor unfettered access to everything.

What would you think about calling it `full_access?` or `unrestricted?`? Or maybe another option?

Reply:
Makes sense - I like `unrestricted?` .. or `bypass?` (just thinking of another option)

Reply:
I'm a fan of `unrestricted?`. I'll go with that. Thanks!